templates/commands/goldentest/new_test_cli_test.go

```diff
@@ -238,3 +239,74 @@ builtin_vars:
 		})
 	}
 }
+
+func TestNewTestFlags_Parse(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name    string
+		args    []string
+		want    NewTestFlags
+		wantErr string
+	}{
+		{
+			name: "all_flags_present",
+			args: []string{
+				"--input", "x=y",
+				"--builtin-var", "_git_tag=my-cool-tag",
+				"--force-overwrite",
+				"--prompt",
+				"new-test",
+				"/a/b/c",
+			},
+			want: NewTestFlags{
+				NewTestName:    "new-test",
+				Location:       "/a/b/c",
+				Inputs:         map[string]string{"x": "y"},
+				BuiltinVars:    map[string]string{"_git_tag": "my-cool-tag"},
+				ForceOverwrite: true,
+				Prompt:         true,
+			},
+		},
+		{
+			name: "default_location",
+			args: []string{
+				"--input", "x=y",
+				"--builtin-var", "_git_tag=my-cool-tag",
+				"--force-overwrite",
+				"--prompt",
+				"new-test",
+			},
+			want: NewTestFlags{
+				NewTestName:    "new-test",
+				Location:       ".",
+				Inputs:         map[string]string{"x": "y"},
+				BuiltinVars:    map[string]string{"_git_tag": "my-cool-tag"},
+				ForceOverwrite: true,
+				Prompt:         true,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var cmd NewTestCommand
+			cmd.SetLookupEnv(cli.MapLookuper(nil))
+
+			err := cmd.Flags().Parse(tc.args)
+			if err != nil || tc.wantErr != "" {
+				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+					t.Fatal(diff)
+				}
+				return
+			}
```

Comment:
Do we want the nil check here? I think we might want to call `DiffErrString` without the surrounding conditional.

Reply:
Thanks for catching it! I was blindly copy the existing test structure in abc goldentest. I sent another PR to fix all of them https://github.com/abcxyz/abc/pull/387